### PR TITLE
Transaction call claims

### PR DESCRIPTION
**Pull request by nwatson22**
This adds some proof claims which it is hoped we can use as a jumping-off point for automatically generating claims from application calls made through the python frontend for kavm.

### CODE REVIEW

File: lib/include/kframework/avm/avm-execution.md [hunk 1]
```diff
@@ -945,52 +945,52 @@ OptIn
...
+//                                  +Int PARAM_APP_OPTIN_FLAT 
+//                                  +Int ((PARAM_MIN_BALANCE_PER_ENTRY +Int PARAM_UINT_MIN_BALANCE) 
+//                                    *Int LOCAL_INTS)
+//                                  +Int ((PARAM_MIN_BALANCE_PER_ENTRY +Int PARAM_BYTES_MIN_BALANCE) 
+//                                    *Int LOCAL_BYTES)
+//           </minBalance>
+//           ...
+//         </account>
+//       </accountsMap>
+//     requires notBool hasOptedInApp(APP_ID, SENDER)
```
Comment (nwatson22) [inline lib/include/kframework/avm/avm-execution.md:993]:
```suggestion
```

File: lib/include/kframework/avm/blockchain.md [hunk 2]
```diff
@@ -972,12 +972,18 @@ The transaction index connects a transaction groups ID to the transaction IDs co
...
-          <txnIndexMapGroupValues> GROUP_INDEX |-> TXN_ID ... </txnIndexMapGroupValues>
+          <txnIndexMapGroupValues> MV </txnIndexMapGroupValues>
         </txnIndexMapGroup>
 
+//  rule [[ getGroupFieldByIdx(GROUP_ID, GROUP_INDEX, FIELD, FIELD_INDEX) => getTxnField(TXN_ID, FIELD, FIELD_INDEX) ]]
+//        <txnIndexMapGroup>
+//          <txnIndexMapGroupKey> GROUP_ID </txnIndexMapGroupKey>
+//          <txnIndexMapGroupValues> GROUP_INDEX |-> TXN_ID ... </txnIndexMapGroupValues>
+//        </txnIndexMapGroup>
+
```
Comment (nwatson22) [inline lib/include/kframework/avm/blockchain.md:986]:
```suggestion
```

File: lib/include/kframework/avm/avm-execution.md [hunk 3]
```diff
@@ -945,52 +945,6 @@ OptIn
 
 // Case 3: needed because of bug?
```
Comment (nwatson22) [inline lib/include/kframework/avm/avm-execution.md:947]:
```suggestion
```

**Comment by geo2a** [comment]
@nwatson22 I've added a lemma that effectively just drops `Sha512_256raw` from proofs. Ideally, we would not want to do that, but rather we'd change the implementation of `getAppAddressBytes` with a lemma. However, I could not manage to make `getAppAddressBytes` to not apply in symbolic cases. We can discuss further when on a call.

**Review by geo2a — APPROVED** [review]

**Comment by geo2a** [comment]
Oops, looks like I had not built K locally for some time. I'll debug the failing claim with the new build of K and update the PR.

**Pull request merged**